src/ChannelServer/World/Map.cs

```diff
@@ -251,6 +251,112 @@ public List<ICombatEntity> GetAttackableEntitiesInRange(ICombatEntity attacker,
+		/// <returns></returns>
+		public List<ICombatEntity> GetAttackableEntitiesInRangeSortedByDistance(Character attacker, Position position, float radius)
+		{
+			var result = new List<ICombatEntity>();
+
+			lock (_monsters)
+			{
+				var entities = _monsters.Values.Where(a => a.Position.InRange2D(position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(position));
+				result.AddRange(entities);
+			}
+
+			lock (_characters)
+			{
+				var entities = _characters.Values.Where(a => a.Position.InRange2D(position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(position));
+				result.AddRange(entities);
+			}
+
+			return result;
+		}
+
+		/// <summary>
+		/// Returns attackable monsters sorted by their distance in the given 
+		/// radius around position.
+		/// </summary>
+		/// <param name="attacker"></param>
+		/// <param name="position"></param>
+		/// <param name="radius"></param>
+		/// <returns></returns>
+		public List<ICombatEntity> GetAttackableEntitiesInRangeAroundEntity(Character attacker, IEntity entity, float radius)
+		{
+			var result = new List<ICombatEntity>();
+
+			lock (_monsters)
+			{
+				var entities = _monsters.Values.Where(a => entity.Handle != a.Handle && a.Position.InRange2D(entity.Position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(entity.Position));
+				result.AddRange(entities);
+			}
+
+			lock (_characters)
+			{
+				var entities = _characters.Values.Where(a => entity.Handle != a.Handle && a.Position.InRange2D(entity.Position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(entity.Position));
+				result.AddRange(entities);
+			}
+
+			return result;
+		}
+
+
+		/// <summary>
+		/// Returns attackable monsters sorted by their distance in the given 
+		/// radius around position.
+		/// </summary>
+		/// <param name="attacker"></param>
+		/// <param name="position"></param>
+		/// <param name="radius"></param>
+		/// <returns></returns>
+		public ICombatEntity GetNearestAttackableEntity(Character attacker, Position position, float radius)
+		{
+			ICombatEntity result;
+
+			lock (_monsters)
+			{
+				result = _monsters.Values.Where(a => a.Position.InRange2D(position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(position)).FirstOrDefault();
+				if (result != null)
+					return result;
+			}
+
+			lock (_characters)
+			{
+				result = _characters.Values.Where(a => a.Position.InRange2D(position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(position)).FirstOrDefault();
+			}
+
+			return result;
+		}
+
+		/// <summary>
+		/// Returns nearest attackable entity in the given radius around position.
+		/// </summary>
+		/// <param name="position"></param>
+		/// <param name="radius"></param>
+		/// <returns></returns>
+		public List<ICombatEntity> GetAttackableEntitiesInRangeSortedByDistance(ICombatEntity attacker, Position position, float radius)
```

Comment:
Isn't this the same as the other `GetAttackableEntitiesInRangeSortedByDistance`?

Reply:
That sounds about right, no need for the other one since this one accepts ICombatEntity as the attacker.